### PR TITLE
Avoid raise condition in os.environ dict (TESTING IN PROGRESS)

### PR DESCRIPTION
**Pull request by ilia-tsyplenkov**
https://github.com/reportportal/client-Python/issues/94
After testing going to provide this fix in base master (possible it will be better to create a separate PR)

### CODE REVIEW

File: reportportal_client/service.py [hunk 1]
```diff
@@ -85,6 +85,25 @@ def _get_messages(data):
     return error_messages
 
 
+def _perform_reliably(request, **kwargs):
+    """Fix for https://github.com/reportportal/client-Python/issues/94
+    Attempt to perform provided HTTP request with kwargs,
+    Ingore KeyError exception first HTTP_REQUEST_RETRY_COUNT times.
+    Return a response or propagate last exception if HTTP_REQUEST_RETRY_COUNT exceeded.
+    """
+
```
Comment (iivanou) [inline reportportal_client/service.py:94]: No blank lines allowed after function docstring

File: reportportal_client/service.py [hunk 2]
```diff
@@ -85,6 +85,25 @@ def _get_messages(data):
     return error_messages
 
 
+def _perform_reliably(request, **kwargs):
+    """Fix for https://github.com/reportportal/client-Python/issues/94
+    Attempt to perform provided HTTP request with kwargs,
```
Comment (iivanou) [inline reportportal_client/service.py:90]: 1 blank line required between summary line and description

File: reportportal_client/service.py [hunk 3]
```diff
@@ -85,6 +85,25 @@ def _get_messages(data):
     return error_messages
 
 
+def _perform_reliably(request, **kwargs):
+    """Fix for https://github.com/reportportal/client-Python/issues/94
```
Comment (iivanou) [inline reportportal_client/service.py:89]: First line should end with a period (not '4')

File: reportportal_client/service.py [hunk 4]
```diff
@@ -85,6 +85,25 @@ def _get_messages(data):
...
+
+    res = None
+    for _ in range(HTTP_REQUEST_RETRY_COUNT):
+        try:
+            res = request(**kwargs)
+            break
+        except KeyError:
+            pass
+    else:
+        raise  # noqa: pylint: disable=misplaced-bare-raise
```
Comment (iivanou) [inline reportportal_client/service.py:103]: You should not ignore this error. You may raise an incorrect exception in some cases.

```
    error = None
    for attempt in range(HTTP_REQUEST_RETRY_COUNT):
        try:
            return request(**kwargs)
        except KeyError as err:
            error = err
    raise error
```

File: reportportal_client/service.py (hunk 2 above)
Comment (ilia-tsyplenkov) [reply reportportal_client/service.py:91]: Done

File: reportportal_client/service.py (hunk 4 above)
Comment (ilia-tsyplenkov) [reply reportportal_client/service.py]: Good point. Thanks

File: reportportal_client/service.py [hunk 5]
```diff
@@ -85,6 +86,27 @@ def _get_messages(data):
...
 
+def _perform_reliably(request, **kwargs):
+    """Fix for https://github.com/reportportal/client-Python/issues/94.
+
+    Attempt to perform provided HTTP request with kwargs,
+    Ingore KeyError exception first HTTP_REQUEST_RETRY_COUNT times.
+    Return a response or propagate last exception
+        if HTTP_REQUEST_RETRY_COUNT exceeded.
+    """
+    res = None
```
Comment (iivanou) [inline reportportal_client/service.py:97]: Why did not you pick up the requested code block? It contains 4 lines less than yours.

```
    error = None
    for attempt in range(HTTP_REQUEST_RETRY_COUNT):
        try:
            return request(**kwargs)
        except KeyError as err:
            error = err
    raise error
```

Comment (ilia-tsyplenkov) [reply reportportal_client/service.py]: Agree, thanks

**Pull request merged**